components/src/components/elements/included-features/Details.tsx

```diff
@@ -69,43 +77,84 @@ export const Details = ({
     }
 
     if (priceBehavior === "pay_in_advance" && typeof allocation === "number") {
-      return `${formatNumber(allocation)} ${getFeatureName(feature, allocation)}`;
+      return (
+        <>
+          {formatNumber(allocation)} {getFeatureName(feature, allocation)}
+        </>
+      );
     }
 
     if (priceBehavior === "pay_as_you_go" && typeof price === "number") {
-      return `${formatCurrency(price, currency)} ${t("per")} ${getFeatureName(feature, 1)}`;
+      return (
+        <>
+          {formatCurrency(price, currency)} {t("per")}{" "}
+          {packageSize > 1 && <>{packageSize} </>}
+          {getFeatureName(feature, packageSize)}
+        </>
+      );
     }
 
     if (priceBehavior === "overage" && typeof softLimit === "number") {
-      return `${formatNumber(softLimit)} ${getFeatureName(feature, softLimit)}`;
+      return (
+        <>
+          {formatNumber(softLimit)} {getFeatureName(feature, softLimit)}
+        </>
+      );
     }
 
     if (!priceBehavior && typeof allocation === "number") {
-      return `${formatNumber(allocation)} ${getFeatureName(feature, allocation)}`;
+      return (
+        <>
+          {formatNumber(allocation)} {getFeatureName(feature, allocation)}
+        </>
+      );
     }
 
     if (!priceBehavior) {
       return t("Unlimited", { item: getFeatureName(feature) });
     }
-  }, [t, allocation, feature, price, priceBehavior, currency, softLimit]);
+  }, [
+    t,
+    allocation,
+    feature,
+    price,
+    priceBehavior,
+    currency,
+    packageSize,
+    softLimit,
+  ]);
 
   const usageText = useMemo(() => {
     if (!feature) {
       return;
     }
 
-    let acc: string | undefined;
+    const acc: React.ReactElement[] = [];
+    let index = 0;
     if (
       priceBehavior === "pay_in_advance" &&
       typeof data.company?.plan?.planPeriod === "string" &&
       typeof price === "number"
```

Comment:
Not a blocker, but we have a lot of this sort of logic in and around this PR. If we see a chance to extract a named function that's easy to write tests for, we ought to do that imo. At least for someone new like me, it can also serve as documentation of the little bit of business logic behind what we choose to render.